### ISSUE TITLE
CRLF line ends cause issues

### ISSUE THREAD

**Issue by scop** [keith-hall/http-request-response-syntax#4]
HTTP uses CRLF line ends, but it appears this syntax works properly only with LF only (at least for me on Linux).

One example reproducer:
```shellsession
curl -i https://www.iki.fi | bat -l http
```
...results in the body not highlighted as HTML, whereas with
```shellsession
curl -i https://www.iki.fi | sed -e 's/\r//' | bat -l http
```
...it does.

Not sure if it trips up anything else besides the content type/embed detection.

I tried simply replacing `\n` with `\r?\n` in the syntax, but it didn't help.

**Comment by keith-hall** [comment]
Thanks for reporting. I've pushed a commit (556aa14f8c909195f28a20cd510013ee6283d9a2) which adds a variable called `end_of_headers`. Changing this to `(?:^$\n?)` (i.e. making the `\n` optional) fixes the problem.

But this is really a problem with `bat`/`syntect`. Sublime Text normalizes line endings to `\n` so `\r` would never be encountered by plugin code or a syntax definition. `syntect` provides a [LinesWithEndings](https://docs.rs/syntect/4.6.0/syntect/util/struct.LinesWithEndings.html) iterator, but this isn't used by `bat` and doesn't strip the carriage returns anyway.
There are cases where it is useful that `bat` doesn't strip carriage returns, like for `--show-all` highlighting. So I guess the quick fix is to update the HTTP Request/Response syntax definition as mentioned above, but I feel that the underlying issue in `bat` and `syntect` should also be resolved...

**Comment by scop** [comment]
Thanks for taking a look. However the recipe doesn't seem to fix it for me for the test case in the initial comment with bat.
```
$ grep end_of_headers: assets/syntaxes/02_Extra/http-request-response/http-request-response.sublime-syntax
  end_of_headers: (?:^$\n?)
$ assets/create.sh ; cargo install --path .
[...]
$ curl -i https://www.iki.fi | bat -l http  # HTML does not get highlighted properly
```

**Comment by keith-hall** [comment]
You're right, my apologies - I made a local copy of the `sed`ed version and the original and apparently tested against the one with carriage returns removed by mistake.

The correct fix is to use `content_type_sep: (?=;|\r?$)` and `end_of_headers: (?:^\r?$\n)` variables. The `?` after `\n` seems to make no difference. As `\r` is part of the HTTP RFC, let's just fix it here for now and worry about upstream fixes later :)
I've pushed a commit with this fix 93b93261bcd8b4fbb38d7e92d3cf3734f0814e31.
